File: app/src/main/java/com/vdovin/spacex/base/BaseFragment.java
package com.vdovin.spacex.base;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.View;

import butterknife.ButterKnife;
import dagger.android.support.AndroidSupportInjection;

public class BaseFragment extends Fragment {

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        AndroidSupportInjection.inject(this);
        super.onViewCreated(view, savedInstanceState);
        ButterKnife.bind(this, view);
    }
}

File: app/src/main/java/com/vdovin/spacex/base/BaseFragment.kt
package com.vdovin.spacex.base

import androidx.fragment.app.Fragment

abstract class BaseFragment<P : BasePresenter<*>> : Fragment() {
    abstract val presenter : P

    override fun onStart() {
        super.onStart()
        presenter.start()
    }

    override fun onStop() {
        super.onStop()
        presenter.stop()
    }
}